src/ispell_wrapper.rs

```
@@ -0,0 +1,82 @@
+use unicode_normalization::UnicodeNormalization;
+use unicode_normalization::char::is_combining_mark;
+use ispell;
+use errors::{Result, ResultExt};
+
+pub struct SpellCheck {
+    aspell: ispell::SpellChecker,
+    nb_replace: u32,
+    nb_error: u32,
+}
+impl SpellCheck {
+    pub fn new() -> Result<Self> {
+        Ok(SpellCheck {
+               aspell: ispell::SpellLauncher::new().aspell()
+                   .dictionary("fr")
+                   .launch()?,
+               nb_replace: 0,
+               nb_error: 0,
+           })
+    }
+
+    pub fn nb_error(&self) -> u32 {
+        self.nb_error
+    }
+
+    pub fn nb_replace(&self) -> u32 {
+        self.nb_replace
+    }
+
+    pub fn add_word(&mut self, new_word: &str) -> Result<()> {
+        Ok(self.aspell.add_word(new_word)?)
+    }
+
+    pub fn check(&mut self, name: &str) -> Result<String> {
+
+        let errors_res = self.aspell.check(name);
+        let mut new_name = name.to_string();
+        if let Err(e) = errors_res {
+            print!("{:?}", e);
+            println!(" ({})", name);
+            self.nb_error += 1;
+            return Ok(new_name);
+        }
+
+        let misspelt_errors = errors_res.chain_err(|| "Could not perform check using aspell")?;
```

Just keeping that to fail explicitly